### PR TITLE
feat: new lacework_alert_channel_gcp_pub_sub resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Add new resource lacework_alert_channel_gcp_pub_sub

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

**Comment by scottford-lw** [comment]
Great work @dmurray-lacework 👍

**Review by scottford-lw — APPROVED** [review]
👍

File: lacework/resource_lacework_alert_channel_gcp_pub_sub.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"client_email": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"private_key": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_gcp_pub_sub.go:64]: This element **must** be marked as sensitive. Please check how this was implemented here https://github.com/lacework/terraform-provider-lacework/blob/master/lacework/resource_lacework_integration_gcp_cfg.go#L56-L73

File: website/docs/r/alert_channel_gcp_pub_sub.html.markdown [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Alert Channel integration name.
+* `project_id` - (Required) The name of the Gcp Project.
+* `topic_id` - (Required) The id of the Gcp Pub Sub Topic.
```
Comment (afiune) [inline website/docs/r/alert_channel_gcp_pub_sub.html.markdown:37]: ID should be capitalized.
Google's doc refer to Pub/Sub topic: https://cloud.google.com/pubsub

```suggestion
* `topic_id` - (Required) The ID of the Google Cloud Pub/Sub topic.
```

File: website/docs/r/alert_channel_gcp_pub_sub.html.markdown [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Alert Channel integration name.
+* `project_id` - (Required) The name of the Gcp Project.
```
Comment (afiune) [inline website/docs/r/alert_channel_gcp_pub_sub.html.markdown:36]:
```suggestion
* `project_id` - (Required) The name of the Google Cloud Project.
```

File: website/docs/r/alert_channel_gcp_pub_sub.html.markdown [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+resource "lacework_alert_channel_gcp_pub_sub" "example" {
+  name       = "S3 Data Export"
+  project_id = "my-sample-project-191923"
+  topic_id   = "mytopic"
+  credentials {
+    client_id = "client_id"
+    client_email = "foo@example.iam.gserviceaccount.com"
+    private_key = "priv_key"
+    private_key_id = "p_key_id"
+  }
```
Comment (afiune) [inline website/docs/r/alert_channel_gcp_pub_sub.html.markdown:27]: Could we please update three things here, the topic and project ids should be something meaningful. Also, let's use the same docs on the other resources for the `credentials` field here 👆🏽

**Review by afiune — CHANGES_REQUESTED** [review]
Some docs nitpick but one "must change" comment.

File: lacework/resource_lacework_alert_channel_gcp_pub_sub.go [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+						},
+						"private_key": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								if d.HasChanges(
+									"name", "resource_level", "resource_id", "org_level", "enabled",
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_gcp_pub_sub.go:68]:
```suggestion
									"name", "project_id", "topic_id", "org_level", "enabled",
```

**Review by afiune — APPROVED** [review]
tenor-231096426

**Pull request merged**